internal/clientconn/conninfo/conn_info.go

```diff
@@ -91,6 +112,22 @@ func (connInfo *ConnInfo) SetMetadataRecv() {
 	connInfo.metadataRecv = true
 }
 
+// BypassBackendAuth marks the connection as not requiring backend authentication.
+func (connInfo *ConnInfo) BypassBackendAuth() {
+	connInfo.rw.Lock()
+	defer connInfo.rw.Unlock()
+
+	connInfo.bypassBackendAuth = true
+}
+
+// GetBypassBackendAuth returns whether the connection requires backend authentication.
+func (connInfo *ConnInfo) GetBypassBackendAuth() bool {
+	connInfo.rw.RLock()
+	defer connInfo.rw.RUnlock()
+
+	return connInfo.bypassBackendAuth
+}
```

Comment:
Note how other methods are called `X` and `SetX`, not `GetX` and `X`.

Reply:
Changed!